src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -899,8 +899,10 @@ public override AwaitExpressionInfo GetAwaitExpressionInfo(AwaitExpressionSyntax
                 throw new ArgumentException("node.Kind==" + node.Kind());
             }
 
-            var bound = GetUpperBoundNode(node);
-            BoundAwaitableInfo awaitableInfo = (((bound as BoundExpressionStatement)?.Expression ?? bound) as BoundAwaitExpression)?.AwaitableInfo;
+            BoundAwaitExpression boundAwait = (BoundAwaitExpression)GetLowerBoundNode(node);
```

Comment:
>GetLowerBoundNode [](http://example.com/codeflow?start=68&length=17)

I do not expect any changes in this function except a change from using GetUpperBoundNode to using GetLowerBoundNode. Our assumptions of what might happen do not change

Reply:
Could you explain what you mean?

Reply:
I mean ```(((bound as BoundExpressionStatement)?.Expression ?? bound) as BoundAwaitExpression)?.AwaitableInf``` shouldn't change. Conditional access shouldn't be eliminated, asserts should not be added. Only this line ```var bound = GetUpperBoundNode(node);``` should change in this function. Othe lines should not be touched, even to change formatting (we don't welcome style-only changes, BTW).

Reply:
In my eyes this is wrong because we should never get a `BoundExpressionStatement` here - but it is your code.